### PR TITLE
feat: allow endpoint to be submitted as a func as well as static string

### PR DESCRIPTION
**Pull request by chanind**
This PR allows the endpoint to be sent as a function, so we can change it if we discover we're hitting the wrong endpoint on the client (china vs global). For https://github.com/ef-carbon/nightingale-mobile/pull/597

### CODE REVIEW

File: lib/Provider.ts [hunk 1]
```diff
@@ -15,11 +15,13 @@ import fetchStart from '@fetch/start';
 import queryFormations from '@query/formations';
 import queryStart from '@query/start';
 
+type endpointUrlFunc = () => UrlAccepted;
```
Comment (leethree) [inline lib/Provider.ts]: type should be CapitalCase. I suggest the name to be `UrlThunk`

File: lib/Provider.ts [hunk 2]
```diff
@@ -62,7 +64,7 @@ export interface IOptions<DataImmutable, DataMutable extends DataImmutable = Dat
 
 export class Provider<DataImmutable, DataMutable extends DataImmutable = DataImmutable>
   implements ICacheProvider<DataImmutable, FormationWalkOptionsInit> {
-  readonly endpoint: IUrlImmutable;
+  readonly getEndpoint: () => IUrlImmutable;
```
Comment (leethree) [inline lib/Provider.ts]: declare it as a class method instead of a property.

**Comment by leethree** [comment]
does it work if you just upgrade `babel`?

**Comment by leethree** [comment]
I fixed the build issue in #5 instead

**Comment by chanind** [comment]
#5 doesn't really fix things, because if you regenerate yarn.lock from package.json everything is broken. This PR locks the package.json versions as well so it won't break in the future

**Pull request merged**